app/services/change_schedule.rb

```diff
@@ -0,0 +1,174 @@
+# frozen_string_literal: true
+
+class ChangeSchedule
+  include ActiveModel::Model
+  include ActiveModel::Attributes
+  include ActiveModel::Validations::Callbacks
+  include Participants::ProfileAttributes
+
+  attribute :cpd_lead_provider
+  attribute :participant_id
+  attribute :reason
+  attribute :course_identifier
+  attribute :schedule_identifier
+  attribute :cohort
+
+  alias_attribute :user_profile, :participant_profile
+  delegate :participant_profile_state, to: :participant_profile, allow_nil: true
+  delegate :lead_provider, to: :cpd_lead_provider, allow_nil: true
+
+  validates :schedule, presence: { message: I18n.t(:invalid_schedule) }
+  validate :not_already_withdrawn
+  validate :validate_new_schedule_valid_with_existing_declarations
+  validate :validate_provider
+  validate :validate_permitted_schedule_for_course
+  validate :validate_cannot_change_cohort
+  validate :schedule_valid_with_pending_declarations
+
+  def call
+    return if invalid?
+
+    ActiveRecord::Base.transaction do
+      ParticipantProfileSchedule.create!(participant_profile:, schedule:)
+      participant_profile.update_schedule!(schedule)
+
+      if relevant_induction_record
+        Induction::ChangeInductionRecord.call(
+          induction_record: relevant_induction_record,
+          changes: {
+            schedule:,
+          },
+        )
+      end
+    end
+
+    relevant_induction_record_for_profile(participant_profile)
+  end
+
+  def participant_profile
+    @participant_profile ||= ParticipantProfileResolver
+                               .call(
+                                 participant_identity:,
+                                 course_identifier:,
+                                 cpd_lead_provider:,
+                               )
+  end
+
+  def alias_search_query
+    Finance::Schedule
+      .where("identifier_alias IS NOT NULL")
+      .where(identifier_alias: schedule_identifier, cohort:)
+  end
+
+  def schedule
+    @schedule ||= Finance::Schedule
+      .where(schedule_identifier:, cohort:)
+      .or(alias_search_query)
+      .first
+  end
+
+  def valid_courses
+    ValidCoursesResolver.call(
+      course_identifier:,
+    )
+  end
+
+private
+
+  def cohort
+    @cohort ||= if super
+                  Cohort.find_by(start_year: super)
+                else
+                  Cohort.current
+                end
+  end
+
+  def relevant_induction_record
+    return if user.blank? || participant_profile.blank?
+
+    participant_profile
```

Comment:
could we reuse the methods we have already on ECF profiles to look this up?

Reply:
@cwrw All done tks 👍 😄